plugins/out_s3/s3.c

```diff
@@ -2132,11 +2161,37 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
                                     event_chunk->tag,
                                     flb_sds_len(event_chunk->tag));
 
+    if (upload_file == NULL) {
+        /* unpack msgpack */
+        msgpack_unpacked_init(&result);
+
+        /* Get the first record timestamp */
+        while (msgpack_unpack_next(&result,
+                                   event_chunk->data,
+                                   event_chunk->size, &off) == MSGPACK_UNPACK_SUCCESS) {
+            flb_time_pop_from_msgpack(&tms, &result, &obj);
+            if (&tms.tm.tv_sec != 0) {
+                break;
+            }
+        }
+
+        msgpack_unpacked_destroy(&result);
+    }
+
     /* Specific to unit tests, will not get called normally */
     if (s3_plugin_under_test() == FLB_TRUE) {
-        unit_test_flush(ctx, upload_file,
-                        event_chunk->tag, flb_sds_len(event_chunk->tag),
-                        chunk, chunk_size, m_upload_file);
+        if (!upload_file) {
+            unit_test_flush(ctx, upload_file,
+                            event_chunk->tag, flb_sds_len(event_chunk->tag),
+                            chunk, chunk_size,
+                            m_upload_file, &tms.tm.tv_sec);
+        }
+        else {
+            unit_test_flush(ctx, upload_file,
+                            event_chunk->tag, flb_sds_len(event_chunk->tag),
+                            chunk, chunk_size,
+                            m_upload_file, upload_file->first_log_time);
+        }
```

Comment:
Please put this logic into the unit_test_flush function. you should always pass in `chunk_first_log_time` see other comments.

Reply:
make sense